src/Storages/System/StorageSystemUserDefinedFunctions.cpp

```diff
@@ -0,0 +1,188 @@
+#include <Storages/System/StorageSystemUserDefinedFunctions.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypeEnum.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnArray.h>
+#include <Interpreters/Context.h>
+#include <Common/ExternalLoaderStatus.h>
+#include <Functions/UserDefined/ExternalUserDefinedExecutableFunctionsLoader.h>
+#include <Functions/UserDefined/UserDefinedExecutableFunction.h>
+#include <Processors/Sources/ShellCommandSource.h>
+#include <Common/Exception.h>
+
+
+namespace DB
+{
+
+StorageSystemUserDefinedFunctions::StorageSystemUserDefinedFunctions(
+    const StorageID & storage_id_, ColumnsDescription columns_description_)
+    : IStorageSystemOneBlock(storage_id_, std::move(columns_description_))
+{
+}
+
+ColumnsDescription StorageSystemUserDefinedFunctions::getColumnsDescription()
+{
+    return ColumnsDescription
+    {
+        // ===== System/Non-Config Fields (External Loader metadata) =====
+        {"name", std::make_shared<DataTypeString>(),
+            "Function name (UDF identifier)."},
+        {"status", std::make_shared<DataTypeEnum8>(
+            DataTypeEnum8::Values{
+                {"SUCCESS", 0},
+                {"FAILED", 1}
+            }),
+            "Loading status. Possible values: "
+            "SUCCESS — UDF loaded and ready to use, "
+            "FAILED — UDF failed to load (see loading_error_message for details)."},
+        {"loading_error_message", std::make_shared<DataTypeString>(),
+            "Detailed error message when loading fails. Empty if loaded successfully."},
+        {"last_loading_time", std::make_shared<DataTypeDateTime>(),
+            "Timestamp when the last loading attempt started."},
+        {"last_successful_update_time", std::make_shared<DataTypeDateTime>(),
+            "Timestamp of last successful load or update."},
+        {"loading_duration", std::make_shared<DataTypeFloat32>(),
+            "Time spent loading the UDF, in seconds."},
+        {"error_count", std::make_shared<DataTypeUInt64>(),
```

Comment:
`error_count` is a slightly weird field. Does it refer to the number of errors during a single load attempt or the total number of failed load attempts? 

Consider removing this field. To debug a UDF, really only the last error message will be sufficient in my view.